foreshadow/parallelprocessor.py

```diff
@@ -403,12 +468,16 @@ def fit_transform(self, X, y=None, **fit_params):
             )
             for name, trans, cols, weight in self._iter()
         )
-
         if not result:
             # All transformers are None
             return X[[]]
 
         Xs, transformers = zip(*result)
+        if update_column_sharer:
+            self._update_original_column_sharer(column_sharer, transformers)
+            self._update_transformers_with_updated_column_sharer(
```

Comment:
You are probably right. I was in a train of thought that we need to update CS everywhere. Let me verify that the CS instance is auto-updated in the transformers.

Reply:
The reason that I have this line is because the `transformers` extracted from the `result` are returned from the parallel process, which means that are deep copies of the original transformers (and in turn copies of the original column_sharer). The cs instances have different object ids comparing to the original one. Given that we are updating the `self.transformer_list` using the `transformers` in the next line with `self._update_transformer_list(transformers)`, we will need to update the cs instances in the `transformers` first.